### PR TITLE
global: release-checklist

### LINKED ISSUE #159: global: release checklist step 1-3
- Run [the release checklist](https://github.com/inveniosoftware/invenio/wiki/Release-preparation-checklist) step 1-3.
- E.g. remove GPLv2 and ``experimental`` note from the README.rst.

### PR DESCRIPTION
**Pull request by hjhsalo**
* Updated module and general file structure according to latest
  cookiecutter-invenio-module.
  (closes #159)
* Updated copyright to 2017 where applicable
* Fixed typos

Signed-off-by: Harri Hirvonsalo <harri.hirvonsalo@cern.ch>

### CODE REVIEW

**Comment by hjhsalo** [comment]
Can the reviewer (or someone else) comment on `tests/conftest.py`. I think it is not according to https://github.com/inveniosoftware/invenio/wiki/Release-preparation-checklist : 
> Check instance_path and app fixtures. Should be using @pytest.yield_fixture and app fixture should create application context.

Also is the list in `.editorconfig:known_third_party` contains more modules than what is included in `requirements-devel.txt`. Is this ok as https://github.com/inveniosoftware/invenio/wiki/Release-preparation-checklist says the following:
> List of modules in.editorconfig:known_third_party should match 1-to-1 all devel requirements.

File: README.rst [hunk 1]
```diff
@@ -40,8 +40,10 @@
 .. image:: https://img.shields.io/github/license/inveniosoftware/invenio-records-rest.svg
         :target: https://github.com/inveniosoftware/invenio-records-rest/blob/master/LICENSE
 
-
 REST API for invenio-records.
 
-* Free software: GPLv2 license
-* Documentation: https://invenio-records-rest.readthedocs.io/
+*This is an experimental developer preview release.*
```
Comment (lnielsen) [inline README.rst]: This line can be removed. This is not clear in the checklist that it has to be removed, when we go to beta release.

File: RELEASE-NOTES.rst [hunk 2]
```diff
@@ -9,6 +9,8 @@ About
 
 REST API for invenio-records.
 
+*This is an experimental developer preview release.*
```
Comment (lnielsen) [inline RELEASE-NOTES.rst]: ditto here

**Comment by lnielsen** [comment]
Re. ``.editorconfig``: It's ok to have more modules in ``.editorconfig`` than in ``requirements-devel.txt``. The other way around, i.e. module in ``requirements-devel.txt`` but not in ``.editorconfig`` will cause problems with isort and import sorting order.

Re. ``test/conftest.py`` then it could be improved, but it's too much work to change it in this module I think.

**Pull request merged**